### PR TITLE
Use modelindexer instead of libbeat ES output in tests

### PR DESCRIPTION
**Pull request by axw**
## Motivation/summary

- Introduce package modelindexertest, which provides some test functions related to modelindexer and mocking Elasticsearch `_bulk` request handling
- Stop using libbeat/esleg/eslegclient in tests for encoding events
- In beater integration tests configure the elasticsearch output (which uses modelindexer) rather than using "chanClient" (now removed), exercising more production code. The instrumentation tests have changed a little as a result, as using the elasticsearch output means we now have additional trace events.

## Checklist

~- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)~
~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~
~- [ ] Documentation has been updated~

## How to test these changes

N/A

## Related issues

https://github.com/elastic/beats/pull/31553 broke our tests. We should be using modelindexer anyway.

### CODE REVIEW

File: beater/integration_test.go [hunk 1]
```diff
@@ -32,39 +32,51 @@ import (
...
+	if err != nil {
+		panic(err)
+	}
+	if time.Since(timestamp) < 5*time.Minute {
+		var err error
+		doc, err = sjson.SetBytes(doc, "@timestamp", timestampOverride.Format(timestampFormat))
+		if err != nil {
+			panic(err)
+		}
+		if gjson.GetBytes(doc, "processor.name").String() != "metric" {
```
Comment (stuartnelson3) [inline beater/integration_test.go:59]: these seem like two very helpful packages

**Review by stuartnelson3 — APPROVED** [review]
looks good

File: model/modelindexer/modelindexertest/modelindexertest.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+			panic(err)
+		}
+		var actionType string
+		for actionType = range action {
+		}
+		if !scanner.Scan() {
+			panic("expected source")
+		}
+
+		doc := append([]byte{}, scanner.Bytes()...)
```
Comment (marclop) [inline model/modelindexer/modelindexertest/modelindexertest.go:88]: cool way to copy the slice, I would've done something much longer like checking the length of the scanned bytes, creating a `[]byte` with that length and using `copy`.

File: model/modelindexer/modelindexertest/modelindexertest.go [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+		if !json.Valid(doc) {
```
Comment (marclop) [inline model/modelindexer/modelindexertest/modelindexertest.go:89]: TIL

**Review by marclop — APPROVED** [review]
LGTM! `modelindexertest` is pretty neat.

**Pull request merged**